packages/core/http/core-http-router-server-internal/src/versioned_router/core_versioned_route.ts

```diff
@@ -132,8 +153,14 @@ export class CoreVersionedRoute implements VersionedRoute {
   private getVersion(req: RequestLike): ApiVersion | undefined {
     let version;
     const maybeVersion = readVersion(req, this.enableQueryVersion);
-    if (!maybeVersion && (this.isPublic || this.useDefaultStrategyForPath)) {
-      version = this.getDefaultVersion();
+    if (!maybeVersion) {
+      if (this.useDefaultStrategyForPath) {
+        version = this.getDefaultVersion();
+      } else if (!this.isDev && !this.isPublic) {
+        // When in production, we default internal routes to v1 to allow
+        // gracefully onboarding of un-versioned to versioned routes
+        version = '1';
+      }
```

Comment:
This is the fix

Reply:
nit: I'm always confused between `isDev` and `isDist`. AFAIK...
* `yarn start` => `isDev: true`; `isDist: false`
* CI => `isDev: false`; `isDist: false`
* Production and Cloud/Serverless tests => `isDev: false`; `isDist: true`

Are we OK with just validating `isDev`?

Reply:
That's a good q. The primary intention is this fallback is to target dev time.

> CI => isDev: false; isDist: false

I think CI => `isDev: true`:

https://github.com/elastic/kibana/blob/61c9137a1eeb1548e1878110194abc173fe64724/packages/kbn-test/src/functional_tests/start_servers/start_servers.ts#L53

This should be "on" during CI. If we find an issue we should address it in a followup PR!